test/src/core/FObject.js

```diff
@@ -214,3 +214,49 @@ describe('copyFrom', function() {
     expect(obj.d).toBe(true);
   });
 });
+
+describe('FObject compare', function() {
+  beforeAll(function() {
+    foam.CLASS({
+      name: 'SomeClass',
+      properties: [
+        {
+          name: 'a',
+          value: 1
+        }
+      ]
+    });
+  });
+
+  it('with undefined values', function() {
+    var someObj = SomeClass.create();
+    expect(foam.util.compare(undefined, someObj)).toBe(1);
+    expect(foam.util.compare(someObj, undefined)).toBe(-1);
+  });
+
+  it('with boolean values', function() {
+    var someObj = SomeClass.create();
+
+    // foam.Boolean does a loose comparison. Object is considered true
+    expect(foam.util.compare(true, someObj)).toBe(0);
+    expect(foam.util.compare(someObj, true)).toBe(0);
+
+    // Unlike other comparators, foam.Boolean return -1 when object is in 2nd
+    // param and 1 when object is in 1st param.
+    expect(foam.util.compare(false, someObj)).toBe(-1);
+    expect(foam.util.compare(someObj, false)).toBe(1);
```

Comment:
foam.Boolean behaves slightly differently than the other comparators. Observe that the the comparison function of the other types return 1 when object is the 2nd parameter; the boolean comparator returns a -1 instead.